From 40c36c6b9f30a12ba4dbf6daad37f97e6f5345eb Mon Sep 17 00:00:00 2001
From: LAK132 <no@example.com>
Date: Sun, 9 Oct 2022 22:29:33 +1030
Subject: [PATCH] fix building with meson

---
 CMakeLists.txt     | 2 +-
 src/SDL_internal.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b53699..9bde8df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -352,7 +352,7 @@ endif()

 # General includes
 target_compile_definitions(sdl-build-options INTERFACE "-DUSING_GENERATED_CONFIG_H")
-target_include_directories(sdl-build-options BEFORE INTERFACE "${SDL2_BINARY_DIR}/include" "${SDL2_BINARY_DIR}/include-config-$<LOWER_CASE:$<CONFIG>>")
+target_include_directories(sdl-build-options BEFORE INTERFACE "include" "${SDL2_BINARY_DIR}/include" "${SDL2_BINARY_DIR}/include-config-$<LOWER_CASE:$<CONFIG>>")
 # Note: The clang toolset for Visual Studio does not support the '-idirafter' option.
 if(USE_GCC OR (USE_CLANG AND NOT MSVC_CLANG))
   # !!! FIXME: do we _need_ to mess with CMAKE_C_FLAGS here?
diff --git a/src/SDL_internal.h b/src/SDL_internal.h
index 43e407f..3ffd74d 100644
--- a/src/SDL_internal.h
+++ b/src/SDL_internal.h
@@ -29,7 +29,7 @@
 /* This is for a variable-length array at the end of a struct:
     struct x { int y; char z[SDL_VARIABLE_LENGTH_ARRAY]; };
    Use this because GCC 2 needs different magic than other compilers. */
-#if (defined(__GNUC__) && (__GNUC__ <= 2)) || defined(__CC_ARM) || defined(__cplusplus)
+#if (defined(__GNUC__) && (__GNUC__ <= 2)) || defined(__CC_ARM) || defined(__cplusplus) || defined (_MSC_VER)
 #define SDL_VARIABLE_LENGTH_ARRAY 1
 #else
 #define SDL_VARIABLE_LENGTH_ARRAY
--
2.34.1

