From a0564a14278d629e90b6554c55ff4ddd6eb272f7 Mon Sep 17 00:00:00 2001
From: LAK132 <1386467+LAK132@users.noreply.github.com>
Date: Sat, 5 Nov 2022 22:31:24 +1030
Subject: [PATCH] Disable building shared libs

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f3e69b8..bc72630 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,7 +89,7 @@ else()
 endif()
 
 # Determine if we should build shared libraries on this platform.
-get_cmake_property(EIGEN_BUILD_SHARED_LIBS TARGET_SUPPORTS_SHARED_LIBS)
+# get_cmake_property(EIGEN_BUILD_SHARED_LIBS TARGET_SUPPORTS_SHARED_LIBS)
 
 #############################################################################
 # find how to link to the standard libraries                                #
-- 
2.34.1

